src/zarr/core/sync.py

```diff
@@ -89,6 +90,20 @@ def cleanup_resources() -> None:
 atexit.register(cleanup_resources)
 
 
+def reset_resources_after_fork() -> None:
+    """
+    Ensure that global resources are reset after a fork. Without this function,
+    forked processes will retain invalid references to the parent process's resources.
+    """
+    global loop, iothread, _executor
```

Comment:
strictly speaking, loop and iothread don't need to be global, since they are mutated in-place.